src/client/Client.h

```diff
@@ -638,12 +645,12 @@ class Client : public Dispatcher, public md_config_obs_t {
   virtual void shutdown();
 
   // messaging
-  void handle_mds_map(class MMDSMap *m);
-  void handle_fs_map(class MFSMap *m);
-  void handle_fs_map_user(class MFSMapUser *m);
-  void handle_osd_map(class MOSDMap *m);
+  void handle_mds_map(const MMDSMap::const_ref& m);
```

Comment:
[OT] i think we should have something like `Ref<MMDSMap>` instead of `MMDSMap::const_ref` which forces us to include the header for its definition.

Reply:
Done, please take another look @tchaikov .